### DISCUSSION TITLE
Windows 10

### DISCUSSION THREAD

**Discussion by xraptorx** [htpcBeginner/docker-traefik#14] [Ideas]
I am trying to run this on windows 10 Pro.  Is it best to:

1. Put ubuntu server on a virtual machine and install do ker into the virtual Linux box or

2. Install do ker on windows and from there run this container?  Will that work?

**Comment by riyaadhm** [comment]
I'm no expert, but for what it's worth;
I've tried this in a Linux VM, and in docker on windows. 
I don't use use everything in the compose file; and I've customized it significantly to suit my setup. 

I've preferred using wsl2 - maybe give it a try?

**Comment by gzecchi** [comment]
Hi @xraptorx,
it's pretty much the same because Docker Desktop on Windows 10 will convert the OS into an Hyper-V  (more or less the same as VirtualBox) and executes a Linux VM called "Moby VM" but you don't have direct access to it.

With WSL2 you cannot share local folders to Docker containers (https://docs.docker.com/docker-for-windows/#file-sharing).

However, IMHO, the quickest and easiest solution is to install Docker Desktop with Windows 10.

GZ

**Comment by gzecchi** [comment]
> Hi @xraptorx,
> it's pretty much the same because Docker Desktop on Windows 10 will convert the OS into an Hyper-V (more or less the same as VirtualBox) and executes a Linux VM called "Moby VM" but you don't have direct access to it.
> 
> With WSL2 you cannot share local folders to Docker containers (https://docs.docker.com/docker-for-windows/#file-sharing).
> 
> However, IMHO, the quickest and easiest solution is to install Docker Desktop with Windows 10.
> 
> GZ

Hi @xraptorx,
I've just tested Docker Desktop on Windows 10 Build 2004 (May 2020) and the configuration in WSL2 works smoothly.
GZ
